js/customize-snapshots-admin.js

```diff
@@ -0,0 +1,73 @@
+( function( $ ) {
+	'use strict';
+
+	$( function() {
+		var $link, linkText, linkActions, dataSlug, initializeLink;
+
+		$link = $( '.snapshot-toggle-setting-removal' );
+		linkText = [ 'Remove setting', 'Restore setting' ];
+		linkActions = [ 'remove', 'restore' ];
+		dataSlug = 'cs-action';
+
+		initializeLink = function() {
+			$link.text( linkText[ 0 ] )
+				.data( dataSlug, linkActions[ 0 ] );
+		};
+
+		initializeLink();
+
+		$link.on( 'click', function( event ) {
+			var $clickedLink, $settingDisplay, clickedLinkAction, settingId;
+
+			event.preventDefault();
+
+			$clickedLink = $( this );
+			$settingDisplay = $( this ).parents( 'details' );
+			clickedLinkAction = $( this ).data( dataSlug );
+			settingId = $( this ).attr( 'id' );
+
+			this.isLinkSetToRemoveSetting = function() {
```

Comment:
Humm. You're adding these functions to the jQuery object itself. Maybe better to create a `component` object that you attach the functions to, and then call its methods in a click handler instead? You can see a pattern for how to create a `component` in https://github.com/xwp/wp-customize-snapshots/blob/develop/js/customize-snapshots.js and see also how it gets exported to global scope so that the component can be interacted with by other scripts, or even by you on the console.